packages/client/src/components/clock/clock.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from "react";
+
+export default function Clock() {
+  const [dateState, setDateState] = useState(new Date());
+  useEffect(() => {
```

Comment:
it is a great solution, however keep in mind that JS will update state on an unmounted component if Clock disapepars. Therefore it is recommended to use a cleanup function (the return function of useEffect), or use a custom useIntevral hook (which is not a standard react hook, but you can search it up on google or InduláSch client (there is a Clock component also).